dns.go

```diff
@@ -294,6 +327,16 @@ func (client *Client) dnsRequest(uri string, params string, method string) (json
 	}
 	req.Header = hdr
 
+	q := req.URL.Query()
+
+	for k, vs := range urlValues {
+		for _, v := range vs {
+			q.Add(k, v)
+		}
+	}
+
+	req.URL.RawQuery = q.Encode()
```

Comment:
I'd do that before the `http.NewRequest`

Reply:
why ? req.URL.Query() will overwrite by `http.NewRequest` no ?

Reply:
build the URL, then do the NewRequest instead of doing the NewRequest then modifying it's URL.